client/src/worldgen_driver.rs

```diff
@@ -41,37 +42,31 @@ impl WorldgenDriver {
             // there's no point trying to generate chunks.
             return;
         }
-        // An extra dodeca::BOUNDING_SPHERE_RADIUS is needed here because we need to generate
-        // enough chunks to ensure that the all chunks within chunk_generation_distance can be filled
-        // with world generation, which requires all nodes surrounding its vertex to be in the graph.
-        traversal::ensure_nearby(
-            graph,
-            &view,
-            chunk_generation_distance + dodeca::BOUNDING_SPHERE_RADIUS,
-        );
-        for (node_id, _) in traversal::nearby_nodes(
-            graph,
-            &view,
-            chunk_generation_distance + dodeca::BOUNDING_SPHERE_RADIUS,
-        ) {
-            graph.ensure_node_state(node_id);
-        }
+        let local_to_view = view.local.inverse();
+
+        traversal::ensure_nearby(graph, &view, chunk_generation_distance);
         let nearby_nodes = traversal::nearby_nodes(graph, &view, chunk_generation_distance);
 
-        'nearby_nodes: for &(node, _) in &nearby_nodes {
+        'nearby_nodes: for &(node, ref node_transform) in &nearby_nodes {
+            let node_to_view = local_to_view * node_transform;
             for vertex in Vertex::iter() {
                 let chunk_id = ChunkId::new(node, vertex);
 
                 if !matches!(graph[chunk_id], Chunk::Fresh) {
                     continue;
                 }
 
-                let Some(params) =
-                    common::worldgen::ChunkParams::new(graph.layout().dimension(), graph, chunk_id)
-                else {
+                // Skip chunks beyond the chunk generation distance
```

Comment:
Does `nearby_nodes(..., chunk_generation_distance)` already accomplish this? If it's not intended to, explain in a comment.

Reply:
`nearby_nodes` only returns nodes. We would need a `nearby_chunks` function to be as precise as this.